stdlib/transactions/src/main/ballerina/src/transactions/2pc_participant.bal

```diff
@@ -96,27 +97,27 @@ type RemoteParticipant object {
         // to prepare a participant
         boolean successful = true;
 
-        log:printDebug("Preparing remote participant: " + self.participantId);
+        log:printDebug(io:sprintf("Preparing remote participant: %s", self.participantId));
```

Comment:
Even with this approach, the argument is evaluated, right? We need to pass a function pointer. 

e.g.,
```suggestion
        log:printDebug(() => io:sprintf("Preparing remote participant: %s", self.participantId));
```

or
```suggestion
        log:printDebug(function () {
            io:sprintf("Preparing remote participant: %s", self.participantId)
        });
```

Reply:
fixed